src/libcore/str/mod.rs

```diff
@@ -337,6 +344,7 @@ fn unwrap_or_0(opt: Option<&u8>) -> u8 {
 /// Reads the next code point out of a byte iterator (assuming a
 /// UTF-8-like encoding).
 #[unstable(feature = "core")]
+#[inline]
```

Comment:
It is intentional to allow llvm constant folding away parts of the function, which was needed to make generated code similar to the old optimized codepaths.

If there is a better solution to this, I don't know it. :(